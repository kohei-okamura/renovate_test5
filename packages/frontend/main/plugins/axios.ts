/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin } from '@nuxtjs/composition-api'
import { AxiosResponse } from 'axios'
import { option, Option } from 'ts-option'
import { errors } from '~/models/errors'
import { HttpStatusCode } from '~/models/http-status-code'
import { removeNull } from '~/support/utils/remove-null'
import { stringifyQueryParams } from '~/support/utils/stringify-query-params'

export type Cause = any

export interface AxiosErrorWrapper {
  onBadRequest (f: (errors: Dictionary<string[]>) => void): AxiosErrorWrapper
  onError (f: (cause: Cause, statusCode?: number) => void): AxiosErrorWrapper
  onForbidden (f: (cause: Cause) => void): AxiosErrorWrapper
  onNotFound (f: (cause: Cause) => void): AxiosErrorWrapper
  onUnauthorized (f: (cause: Cause) => void): AxiosErrorWrapper
}

class AxiosErrorWrapperImpl implements AxiosErrorWrapper {
  private readonly response: Option<AxiosResponse>
  private readonly statusCode?: number

  constructor (private cause: Cause) {
    this.response = option(cause).flatMap<AxiosResponse>(x => option(x.response))
    this.statusCode = this.response.flatMap(x => option(x.status)).orNull ?? undefined
  }

  onBadRequest (f: (errors: Dictionary<string[]>) => void): AxiosErrorWrapper {
    if (this.statusCode === HttpStatusCode.BadRequest) {
      const errors = this.response
        .flatMap(response => option(response.data))
        .flatMap(data => option(data.errors))
        .getOrElse(() => ({}))
      f(errors)
    }
    return this
  }

  onError (f: (cause: Cause, statusCode?: number) => void): AxiosErrorWrapper {
    f(this.cause, this.statusCode)
    return this
  }

  onForbidden (f: (cause: Cause) => void): AxiosErrorWrapper {
    return this.onGenericError(HttpStatusCode.Forbidden, f)
  }

  onNotFound (f: (cause: Cause) => void): AxiosErrorWrapper {
    return this.onGenericError(HttpStatusCode.NotFound, f)
  }

  onUnauthorized (f: (cause: Cause) => void): AxiosErrorWrapper {
    return this.onGenericError(HttpStatusCode.Unauthorized, f)
  }

  private onGenericError (statusCode: number, f: (cause: Cause) => void): AxiosErrorWrapper {
    if (this.statusCode === statusCode) {
      f(this.cause)
    }
    return this
  }
}

export interface WrapAxiosErrorFunction {
  (cause: Cause): AxiosErrorWrapper
}

export function $wrapAxiosError (cause: Cause): AxiosErrorWrapper {
  return new AxiosErrorWrapperImpl(cause)
}

export default defineNuxtPlugin(({ $axios, error, redirect, app }, inject) => {
  $axios.defaults.paramsSerializer = stringifyQueryParams
  $axios.onResponse(response => {
    // レスポンスから値が null のデータを取り除く
    response.data = removeNull(response.data)
  })
  $axios.onError(reason => {
    const status = reason.response?.status
    const url = reason.response?.config.url
    const passthroughErrors = reason.response?.config.passthroughErrors
    if (status === HttpStatusCode.NotFound && !passthroughErrors) {
      if (url?.startsWith('/')) {
        error(errors.notFound)
      }
    } else if (status === HttpStatusCode.Unauthorized) {
      app.$globalStore.session.deleteAuth()
      // セッション取得の 401（= 未ログイン）はリダイレクトしない
      if (!url?.includes('/sessions/my')) {
        if (app.$routes.path.value !== '/') {
          redirect(`/?path=${app.$routes.fullPath.value}`)
        }
      }
    } else if (status !== HttpStatusCode.BadRequest && !passthroughErrors) {
      app.$alert.error('エラーが発生しました、管理者へのお問い合わせをお願いします。', reason.stack)
    }
  })
  inject('wrapAxiosError', (cause: Cause) => {
    return $wrapAxiosError(cause)
  })
})
