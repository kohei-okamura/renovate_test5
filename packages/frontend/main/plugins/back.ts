/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin } from '@nuxtjs/composition-api'
import VueRouter, { Route } from 'vue-router'

export type Back = {
  (fallback?: string): Promise<Route>
}

/**
 * ブラウザバックプラグイン.
 */
export default defineNuxtPlugin((context, inject) => {
  const back: Back = (fallback = '/') => {
    const router: VueRouter = context.app.router!
    return window.history.length === 1 ? router.replace(fallback) : new Promise<Route>(() => router.back())
  }
  inject('back', back)
})
