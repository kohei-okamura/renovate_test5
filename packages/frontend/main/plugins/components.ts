/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin } from '@nuxtjs/composition-api'
import Vue from 'vue'
import ZDataCardItem from '~/components/ui/z-data-card-item.vue'
import ZDataCard from '~/components/ui/z-data-card.vue'

/**
 * コンポーネントを利用可能とする.
 *
 * 一部のコンポーネントが VuetifyLoaderPlugin による動的ロードの対象とならないケースに対応する.
 */
export function setupComponents () {
  Vue.component('ZDataCard', ZDataCard)
  Vue.component('ZDataCardItem', ZDataCardItem)
}

/**
 * コンポーネントプラグイン.
 */
export default defineNuxtPlugin(() => {
  setupComponents()
})
