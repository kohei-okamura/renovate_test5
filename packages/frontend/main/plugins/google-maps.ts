/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin } from '@nuxtjs/composition-api'

export default defineNuxtPlugin((_, inject) => {
  inject('google', () => {
    if (window.google) {
      return Promise.resolve(window.google)
    } else {
      return new Promise(resolve => {
        const callback = () => {
          window.google ? resolve(window.google) : setTimeout(callback, 100)
        }
        const key = process.env.googleMapsApiKey ?? ''
        const version = process.env.googleMapsApiVersion ?? ''
        const script = document.createElement('script')
        script.async = true
        script.src = `https://maps.googleapis.com/maps/api/js?key=${key}&v=${version}`
        script.onload = callback
        document.head.appendChild(script)
      })
    }
  })
})
