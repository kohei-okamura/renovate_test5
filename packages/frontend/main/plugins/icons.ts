/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import {
  mdiAccountCogOutline,
  mdiAccountMultiple,
  mdiAccountOutline,
  mdiAccountStarOutline,
  mdiAlertOutline,
  mdiArrangeSendBackward,
  mdiArrowCollapseUp,
  mdiArrowRightBold,
  mdiBankOutline,
  mdiBarcode,
  mdiBell,
  mdiBookEditOutline,
  mdiBookOffOutline,
  mdiBookOpenVariant,
  mdiBookRemoveOutline,
  mdiCakeVariant,
  mdiCalculator,
  mdiCalendar,
  mdiCalendarCheck,
  mdiCalendarClock,
  mdiCalendarMonth,
  mdiCalendarQuestion,
  mdiCalendarRemove,
  mdiCalendarStar,
  mdiCalendarWeek,
  mdiCardAccountDetailsOutline,
  mdiCardsOutline,
  mdiCertificateOutline,
  mdiChartPie,
  mdiCheckCircle,
  mdiCheckCircleOutline,
  mdiCheckUnderline,
  mdiChevronDown,
  mdiChevronLeft,
  mdiChevronRight,
  mdiChevronUp,
  mdiCityVariantOutline,
  mdiClipboardCheckOutline,
  mdiClockOutline,
  mdiClose,
  mdiCloseCircleOutline,
  mdiCloudUpload,
  mdiCogs,
  mdiCommentAlertOutline,
  mdiCurrencyJpy,
  mdiDeleteOutline,
  mdiDomain,
  mdiDotsHorizontalCircleOutline,
  mdiDotsVertical,
  mdiDownload,
  mdiDragHorizontal,
  mdiEmailOutline,
  mdiEmoticonConfusedOutline,
  mdiEyeOffOutline,
  mdiEyeOutline,
  mdiFileDocumentOutline,
  mdiFileExcelOutline,
  mdiFilePdfBox,
  mdiFileTableBoxOutline,
  mdiFileTableOutline,
  mdiFilter,
  mdiFlagOutline,
  mdiFormatListChecks,
  mdiFormatListText,
  mdiHelpBox,
  mdiHelpCircleOutline,
  mdiHumanCane,
  mdiHumanMaleFemale,
  mdiLock,
  mdiMagnify,
  mdiMapMarkerOutline,
  mdiMinusCircleOffOutline,
  mdiMinusCircleOutline,
  mdiNotebookEditOutline,
  mdiNumeric,
  mdiPauseCircleOutline,
  mdiPencil,
  mdiPhone,
  mdiPlaylistEdit,
  mdiPlus,
  mdiPoll,
  mdiPoundBoxOutline,
  mdiRefresh,
  mdiRepeat,
  mdiSend,
  mdiShieldKeyOutline,
  mdiTargetVariant,
  mdiText,
  mdiTimerSand,
  mdiTimetable,
  mdiUndoVariant,
  mdiViewDashboard,
  mdiWalk,
  mdiWalletMembership,
  mdiWalletOutline
} from '@mdi/js'
import { mdiCalendarRange } from '@mdi/js/commonjs/mdi'
import { defineNuxtPlugin } from '@nuxtjs/composition-api'

// @mdi/js には https://material.io/resources/icons/ の accessible に該当するアイコンの定義がない
// どうやら http://accessibleicon.org/ 絡みらしい……がわかりやすさを優先して独自に定義して用いることにする
const accessible = 'M10 4a2 2 0 104 0 2 2 0 10-4 0M19 13v-2c-1.54.02-3.09-.75-4.07-1.83l-1.29-1.43c-.17-.19-.38-.34-.61-.45-.01 0-.01-.01-.02-.01H13c-.35-.2-.75-.3-1.19-.26C10.76 7.11 10 8.04 10 9.09V15c0 1.1.9 2 2 2h5v5h2v-5.5c0-1.1-.9-2-2-2h-3v-3.45c1.29 1.07 3.25 1.94 5 1.95zm-6.17 5c-.41 1.16-1.52 2-2.83 2-1.66 0-3-1.34-3-3 0-1.31.84-2.41 2-2.83V12.1a5 5 0 105.9 5.9h-2.07z'

// アイコン一覧にあるのに何故か @mdi/js に定義がない
const mdiProgressPencil = 'M15.84 10.2L14.83 11.21L12.76 9.18L13.77 8.16C13.97 7.95 14.31 7.94 14.55 8.16L15.84 9.41C16.05 9.62 16.06 9.96 15.84 10.2M8 13.91L12.17 9.72L14.24 11.8L10.08 16H8V13.91M13 2V4C17.39 4.54 20.5 8.53 19.96 12.92C19.5 16.56 16.64 19.43 13 19.88V21.88C18.5 21.28 22.45 16.34 21.85 10.85C21.33 6.19 17.66 2.5 13 2M11 2C9.04 2.18 7.19 2.95 5.67 4.2L7.1 5.74C8.22 4.84 9.57 4.26 11 4.06V2.06M4.26 5.67C3 7.19 2.24 9.04 2.05 11H4.05C4.24 9.58 4.8 8.23 5.69 7.1L4.26 5.67M2.06 13C2.26 14.96 3.03 16.81 4.27 18.33L5.69 16.9C4.81 15.77 4.24 14.42 4.06 13H2.06M7.06 18.37L5.67 19.74C7.18 21 9.04 21.79 11 22V20C9.58 19.82 8.23 19.25 7.1 18.37H7.06Z'

export const $icons = {
  add: mdiPlus,
  addr: mdiMapMarkerOutline,
  admin: mdiShieldKeyOutline,
  alert: mdiAlertOutline,
  amount: mdiPoll,
  back: mdiChevronLeft,
  bank: mdiBankOutline,
  billing: mdiCalculator,
  birthday: mdiCakeVariant,
  blank: '',
  carePlanAuthor: mdiNotebookEditOutline,
  category: mdiCardsOutline,
  certification: mdiCertificateOutline,
  city: mdiCityVariantOutline,
  close: mdiClose,
  completed: mdiCheckCircle,
  comprehensiveService: mdiWalk,
  confirmed: mdiCheckCircle,
  copayLimit: mdiArrowCollapseUp,
  copy: mdiArrangeSendBackward,
  csv: mdiFileTableOutline,
  dashboard: mdiViewDashboard,
  date: mdiCalendar,
  dateEnd: mdiCalendarRemove,
  dateRange: mdiCalendarRange,
  dateStart: mdiCalendarCheck,
  dayOfWeek: mdiCalendarWeek,
  days: mdiCalendarQuestion,
  defrayerCategory: mdiWalletMembership,
  delete: mdiDeleteOutline,
  disableContract: mdiBookOffOutline,
  document: mdiFileDocumentOutline,
  download: mdiDownload,
  dws: accessible,
  dwsNumber: mdiBarcode,
  dwsType: mdiEmoticonConfusedOutline,
  edit: mdiPencil,
  editVariant: mdiPlaylistEdit,
  editContract: mdiBookEditOutline,
  email: mdiEmailOutline,
  employeeNumber: mdiCardAccountDetailsOutline,
  expand: mdiChevronDown,
  filter: mdiFilter,
  fix: mdiCheckUnderline,
  forward: mdiChevronRight,
  headcount: mdiAccountMultiple,
  help: mdiHelpBox,
  id: mdiPoundBoxOutline,
  invisible: mdiEyeOffOutline,
  issuedOn: mdiCalendarStar,
  keyword: mdiMagnify,
  ledger: mdiBookOpenVariant,
  level: mdiPoll,
  ltcs: mdiHumanCane,
  ltcsInsNumber: mdiBarcode,
  month: mdiCalendarMonth,
  more: mdiDotsVertical,
  moveRight: mdiArrowRightBold,
  note: mdiText,
  notification: mdiBell,
  number: mdiPoundBoxOutline,
  objective: mdiFlagOutline,
  office: mdiDomain,
  ownExpenseProgram: mdiFormatListText,
  password: mdiLock,
  pdf: mdiFilePdfBox,
  personalName: mdiAccountOutline,
  problem: mdiCommentAlertOutline,
  provisionReport: mdiClipboardCheckOutline,
  ratio: mdiChartPie,
  recipientNumber: mdiBarcode,
  recurrence: mdiRepeat,
  refresh: mdiRefresh,
  request: mdiAccountStarOutline,
  role: mdiAccountCogOutline,
  save: mdiCloudUpload,
  schedule: mdiCalendarClock,
  scope: mdiTargetVariant,
  score: mdiFileTableBoxOutline,
  send: mdiSend,
  serviceCode: mdiNumeric,
  serviceOption: mdiFormatListChecks,
  settings: mdiCogs,
  sex: mdiHumanMaleFemale,
  shift: mdiTimetable,
  shrink: mdiChevronUp,
  sortable: mdiDragHorizontal,
  staff: mdiAccountOutline,
  statusChecking: mdiProgressPencil,
  statusDisabled: mdiMinusCircleOffOutline,
  statusNotAvailable: mdiMinusCircleOutline,
  statusProgress: mdiDotsHorizontalCircleOutline,
  statusReady: mdiDotsHorizontalCircleOutline,
  statusRejected: mdiCloseCircleOutline,
  statusResolved: mdiCheckCircleOutline,
  statusSuspended: mdiPauseCircleOutline,
  statusUnknown: mdiHelpCircleOutline,
  tel: mdiPhone,
  terminateContract: mdiBookRemoveOutline,
  text: mdiText,
  timeAmount: mdiTimerSand,
  timeframe: mdiClockOutline,
  undo: mdiUndoVariant,
  upload: mdiCloudUpload,
  user: accessible,
  visible: mdiEyeOutline,
  wallet: mdiWalletOutline,
  xlsx: mdiFileExcelOutline,
  yen: mdiCurrencyJpy
}

/**
 * アイコンプラグイン.
 */
export default defineNuxtPlugin((_, inject) => {
  inject('icons', $icons)
})
