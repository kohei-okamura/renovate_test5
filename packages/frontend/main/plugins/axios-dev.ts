/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin } from '@nuxtjs/composition-api'
import { tap } from '@zinger/helpers'
import { stringifyQueryParams } from '~/support/utils/stringify-query-params'
import { stubAxios } from '~~/stubs/axios'

/**
 * リアクティブな値を含むオブジェクトを単純なオブジェクトに変換する.
 */
const sanitize = <T> (x: T): T => JSON.parse(JSON.stringify(x))

/**
 * Axios 開発用設定プラグイン.
 */
export default defineNuxtPlugin(({ $axios }) => {
  stubAxios($axios)
  $axios.interceptors.request.use(config => tap(config, () => {
    window.console.info(`${config.method} ${config.url}`, {
      data: sanitize(config.data ?? {}),
      params: sanitize(config.params ?? {}),
      query: stringifyQueryParams(config.params)
    })
  }))
})
