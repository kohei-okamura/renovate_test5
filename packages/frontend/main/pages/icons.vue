<!--
  - Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
  - UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
  -->
<template>
  <z-page data-page-icons-index :class="$style.root" :breadcrumbs="[]">
    <z-data-table :items="icons" :options="tableOptions">
      <template #item.icon="{ item }">
        <v-icon>{{ item.icon }}</v-icon>
      </template>
    </z-data-table>
  </z-page>
</template>

<script lang="ts">
import { defineComponent, useCssModule } from '@nuxtjs/composition-api'
import { Permission } from '@zinger/enums/lib/permission'
import { keys } from '@zinger/helpers'
import { dataTableOptions } from '~/composables/data-table-options'
import { auth } from '~/middleware/auth'
import { Role } from '~/models/role'
import { $icons } from '~/plugins/icons'

export default defineComponent({
  name: 'RolesIndexPage',
  middleware: [auth(Permission.listRoles)],
  setup () {
    const style = useCssModule()
    const descriptions: Record<keyof typeof $icons, string> = {
      add: '「追加」を表すボタン等に用いる。',
      addr: '「住所」を表す項目等に用いる',
      admin: '「管理者」を表す項目等に用いる。',
      alert: '「警告」を表すメッセージなどに用いる。',
      amount: '「提供量」などの「量」を表す項目等に用いる。',
      back: '「戻る」ボタン等に用いる。',
      bank: '「銀行」に関連する項目等に用いる。',
      billing: '「請求」を表すメニュー等に用いる。',
      birthday: '「生年月日」を表す項目に用いる。',
      blank: 'アイコンが必要な箇所でアイコンを表示したくない場合に用いる。',
      carePlanAuthor: '「居宅介護支援」所謂ケアマネに関する項目等に用いる。',
      category: 'カテゴリや区分を表す項目等に用いる。',
      certification: 'スタッフの「資格」を表す項目等に用いる。',
      city: '団体としての「市町村」「自治体」等を表す項目に用いる。',
      close: '「閉じる」ボタン等に用いる。',
      completed: '「完了」を表すメッセージ等に用いる。',
      comprehensiveService: '「総合事業」を表す項目等に用いる。',
      confirmed: '「確認済」であることを表す項目等に用いる。',
      copayLimit: '利用者負担等の「上限」を表す項目等に用いる。',
      copy: '「コピー」「複製」を表すボタン等に用いる。',
      csv: '「CSV ファイル」を表す項目等に用いる。',
      dashboard: '「ダッシュボード」を表すメニュー等に用いる。',
      date: '「日時」「年月日」など日付関連を値を表す項目等に用いる。',
      dateEnd: '「終了日」「中止日」など期間の終了を表す項目に用いる。',
      dateRange: '「日時」「年月日」など日付関連の範囲を表す項目等に用いる。',
      dateStart: '「契約日」「開始年月日」など期間の開始を表す項目に用いる。',
      dayOfWeek: '「曜日」を表す項目等に用いる。',
      days: '「日数」を表す項目等に用いる。',
      defrayerCategory: '「法別番号」「公費制度」を表す項目等に用いる。',
      delete: '「削除」を表すボタン等に用いる。',
      disableContract: '「契約」の無効化を表すボタン等に用いる。',
      document: '「ドキュメント」を表す項目等に用いる。',
      download: '「ダウンロード」を表すボタン等に用いる。',
      dws: '「障害福祉サービス」「障害者総合支援」関連の値を表す項目等に用いる。',
      dwsNumber: '「受給者証番号」を表す項目に用いる。',
      dwsType: '「障害種別」を表す項目に用いる。',
      edit: '「編集」「更新」を表すボタン等に用いる。',
      editVariant: '「編集」「更新」を表すボタン等に用いる。同じスピードダイアルで `edit` が既に使われている場合用。',
      editContract: '「契約」の編集を表すボタンに用いる。',
      email: '「メールアドレス」を表す項目等に用いる。',
      employeeNumber: 'スタッフの「社員番号」を表す項目に用いる。',
      expand: '「開く」動作を表すボタン等に用いる。',
      filter: '検索フォームを表す場合等に用いる。',
      fix: '「確定」を表すボタン等に用いる。',
      forward: '「進む」ボタン等に用いる。',
      headcount: '「人数」を表す項目等に用いる。',
      help: 'ヘルプメッセージを表示する場合に用いる。',
      id: '「ID」を表す項目等に用いる。',
      invisible: '「表示・非表示」が切り替え可能な場合の「非表示」を表す場合に用いる。',
      issuedOn: '「交付日」を表す項目に用いる。',
      keyword: '「キーワード」を表す項目等に用いる。',
      ledger: '「台帳」を表すメニュー等に用いる。',
      level: '「要介護度」「障害支援区分」など何らかの「程度」を表す項目に用いる。',
      ltcs: '介護保険関連の値を表す項目等に用いる。',
      ltcsInsNumber: '「被保険者証番号」を表す項目に用いる。',
      month: '「サービス提供年月」等の「月」を表す項目に用いる。',
      more: '「もっと見る」のようなアクションを表すボタン等に用いる。',
      moveRight: '「右へ移動させる」のようなアクションを表すボタン等に用いる。',
      note: '「備考」を表す項目等に用いる。',
      notification: '「通知」を表すボタン等に用いる。',
      number: '「番号」を表す項目等に用いる。',
      objective: '「目標」を表す項目等に用いる。',
      office: '「事業所」を表す項目等に用いる。',
      ownExpenseProgram: '「自費サービス」を表す項目等に用いる。',
      password: '「パスワード」を表す項目等に用いる。',
      pdf: '「PDF ファイル」を表す項目等に用いる。',
      personalName: '利用者やスタッフ以外の「氏名」や「担当者名」を表す項目等に用いる。',
      problem: '「解決すべき課題」を表す項目等に用いる。',
      provisionReport: '「予実」を表す項目等に用いる。',
      ratio: '「利用者負担割合」等の「割合」を表す項目に用いる。',
      recipientNumber: '公費等の「受給者番号」を表す項目に用いる。',
      recurrence: '「周期」を表す項目等に用いる。',
      refresh: '「更新」を表す項目等に用いる。',
      request: '計画の「ご本人の希望」等の項目に用いる。',
      role: '「ロール」を表す項目等に用いる。',
      save: '「保存」等を表すボタンに用いる。',
      schedule: '日付と時刻の両方を伴うスケジュールを表す項目等に用いる。',
      scope: '「権限範囲」を表す項目に用いる。',
      score: '請求における「単位」を表す項目に用いる。',
      send: '「送信」を表すボタン等に用いる。',
      serviceCode: '「サービスコード」を表す項目等に用いる。',
      serviceOption: '「サービスオプション」を表す項目等に用いる。',
      settings: '「設定」を表すメニュー等に用いる。',
      sex: '「性別」を表す項目等に用いる。',
      shift: '「勤務シフト」を表す項目等に用いる。',
      shrink: '「開く」に対応する「閉じる」動作を表すボタン等に用いる。',
      sortable: '並び替え可能なリストにおいて並び替えが可能であることを示すアイコンとして用いる。',
      staff: '「スタッフ」を表す項目等に用いる。',
      statusChecking: '「入力中」等の状態を表すアイコンとして用いる。',
      statusDisabled: '「無効」等の状態を表すアイコンとして用いる。',
      statusNotAvailable: '「未作成」等の状態を表すアイコンとして用いる。',
      statusProgress: '「処理中」等の状態を表すアイコンとして用いる。',
      statusReady: '「未確定」等の準備ができている状態を表すアイコンとして用いる。',
      statusRejected: '「無効」等の状態を表すアイコンとして用いる。',
      statusResolved: '「確定」等の状態を表すアイコンとして用いる。',
      statusSuspended: '「保留」等の状態を表すアイコンとして用いる。',
      statusUnknown: '状態がわからないことを表すアイコンとして用いる。',
      tel: '「電話番号」等を表す項目に用いる。',
      terminateContract: '契約の「終了」を表すボタン等に用いる。',
      text: '「備考」等の文章を入力する項目等に用いる。',
      timeAmount: '時間の「量」を表す項目等に用いる。',
      timeframe: '夜間・日中などの「時間帯」を表す項目等に用いる。',
      undo: '確定した請求を未確定に戻す等の「アンドゥ」操作を表すボタン等に用いる。',
      upload: '「アップロード」を表すボタン等に用いる。',
      user: '「利用者」を表す項目等に用いる。',
      visible: '「表示・非表示」が切り替え可能な場合の「表示」を表す場合に用いる。',
      wallet: '「支払い方法」等を表す項目等に用いる。',
      xlsx: '「エクセルファイル」を表す項目等に用いる。',
      yen: '金額を表す項目等に用いる。'
    }
    const icons = keys($icons).map(name => ({
      name,
      icon: $icons[name],
      description: descriptions[name]
    }))
    const tableOptions = dataTableOptions<Role>({
      content: 'アイコン',
      headers: [
        { text: '物理名', value: 'name', class: style.name, align: 'start', width: 100 },
        { text: 'アイコン', value: 'icon', class: style.icon, align: 'center', width: 100 },
        { text: '用途', value: 'description', class: style.description, align: 'start' }
      ]
    })
    return {
      icons,
      tableOptions
    }
  }
})
</script>

<style lang="scss" module>
.root {
  .name, .icon, .description {
    width: auto;
  }

}
</style>
