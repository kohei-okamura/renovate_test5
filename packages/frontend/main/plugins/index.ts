/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { WrapAxiosErrorFunction } from '~/plugins/axios'
import { Back } from '~/plugins/back'
import { $icons } from '~/plugins/icons'
import { UseFetch } from '~/plugins/nuxt-composition-api'
import { Services } from '~/plugins/services'
import { DatetimeService } from '~/services/datetime-service'

export type Plugins = Services & Readonly<{
  $back: Back
  $datetime: DatetimeService
  $google: () => Promise<Google>
  $icons: typeof $icons
  $useFetch: UseFetch
  $wrapAxiosError: WrapAxiosErrorFunction
}>
