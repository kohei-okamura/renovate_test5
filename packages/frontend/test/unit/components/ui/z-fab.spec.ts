/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Wrapper } from '@vue/test-utils'
import Vue from 'vue'
import ZFab from '~/components/ui/z-fab.vue'
import { $icons } from '~/plugins/icons'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

describe('z-fab.vue', () => {
  const { mount } = setupComponentTest()
  let wrapper: Wrapper<Vue>

  beforeAll(() => {
    const propsData = {
      icon: $icons.add
    }
    wrapper = mount(ZFab, { propsData })
  })

  afterAll(() => {
    wrapper.destroy()
  })

  it('should be rendered correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
