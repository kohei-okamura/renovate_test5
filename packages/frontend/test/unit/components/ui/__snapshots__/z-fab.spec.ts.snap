// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-fab.vue should be rendered correctly 1`] = `
<transition-stub
  mode="out-in"
  name="fab-transition"
>
  <button
    class="z-fab v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round theme--dark v-size--default accent"
    type="button"
  >
    <span
      class="v-btn__content"
    >
      <span
        aria-hidden="true"
        class="v-icon notranslate theme--dark"
      >
        <svg
          aria-hidden="true"
          class="v-icon__svg"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          />
        </svg>
      </span>
    </span>
  </button>
</transition-stub>
`;
